sdk/Sdk.Analyzers/BindingTypeNotSupported.cs

```diff
@@ -0,0 +1,133 @@
+    public class BindingTypeNotSupported : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(DiagnosticDescriptors.BindingTypeNotSupported); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext context)
+        {
+            var method = (IMethodSymbol)context.Symbol;
+
+            if (!method.IsFunction(context))
+            {
+                return;
+            }
+
+            var methodParameters = method.Parameters;
+
+            if (method.Parameters.Length == 0)
+            {
+                return;
+            }
+
+            foreach (var parameter in methodParameters)
+            {
+                AnalyzeParameter(context, parameter);
+            }
+        }
+
+        private static void AnalyzeParameter(SymbolAnalysisContext context, IParameterSymbol parameter)
+        {
+            foreach (var attribute in parameter.GetAttributes())
+            {
+                var attributeType = attribute?.AttributeClass;
+                if (!attributeType.IsInputOrTriggerBinding())
+                {
+                    continue;
+                }
+
+                var inputConverterAttributes = GetInputConverterAttributes(context, attributeType);
+                if (inputConverterAttributes.Count == 0)
+                {
+                    continue;
+                }
+
+                var allowConverterFallbackParameterValue = GetAllowConverterFallbackParameterValue(context, attributeType);
+                if (allowConverterFallbackParameterValue is bool allowFallback && allowFallback)
+                {
+                    // If allowConverterFallback is true, we don't need to check for supported types
+                    // because we don't know all of the types that are supported via the fallback
+                    continue;
+                }
+
+                var supportedTypes = GetSupportedTypes(context, inputConverterAttributes);
+                if (supportedTypes.Count <= 0 || supportedTypes.Contains(parameter.Type))
+                {
+                    continue;
+                }
+
+                ReportDiagnostic(context, parameter, attributeType);
+            }
+        }
+
+        private static List<AttributeData> GetInputConverterAttributes(SymbolAnalysisContext context, ITypeSymbol attributeType)
+        {
+            var inputConverterAttributeType = context.Compilation.GetTypeByMetadataName(Constants.Types.InputConverterAttribute);
+            return attributeType.GetAttributes()
+                .Where(attr => attr.AttributeClass.Equals(inputConverterAttributeType))
```

Comment:
Switch to SymbolEqualityComparer instead of Equals.

Example: https://github.com/Azure/azure-functions-dotnet-worker/blob/a625cc1efc49d37611cbc35e88dbd376b073e49a/sdk/Sdk.Generators/FunctionsUtil.cs#L34

Reply:
Same comment for line 91